src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -41,29 +41,105 @@ static const cmatrix_t one_measure =
 //------------------------------------------------------------------------
 // local function declarations
 //------------------------------------------------------------------------
+//------------------------------------------------------------------------
+// Function name: reorder_all_qubits
+// Description: The ordering of the amplitudes in the statevector in this module is 
+//    [n, (n-1),.., 2, 1, 0], i.e., msb is leftmost and lsb is rightmost.
+//    Sometimes, we need to provide a different ordering of the amplitudes, such as 
+//    in snapshot_probabilities. For example, instead of [2, 1, 0] the user requests 
+//    the probabilities of [1, 0, 2].
+//    Note that the ordering in the qubits vector is the same as that of the mps solver,
+//    i.e., qubits are numbered from left to right, e.g., 210
+// Input: orig_probvector - the ordered vector of probabilities
+//        qubits - a list containing the new ordering
+// Returns: new_probvector - the vector in the new ordering
+//    e.g., 011->101 (for the ordering [1, 0, 2]
+//
+//------------------------------------------------------------------------
+template <class T>
+void reorder_all_qubits(const std::vector<T>& orig_probvector, 
+			reg_t qubits, 
+			std::vector<T>& new_probvector);
+uint_t reorder_qubits(const reg_t qubits, uint_t index);
+
 //--------------------------------------------------------------------------
 // Function name: reverse_all_bits
 // Description: The ordering of the amplitudes in the statevector in this module is 
 //    000, 001, 010, 011, 100, 101, 110, 111.
 //    The ordering of the amplitudes in the statevector in Qasm in general is 
 //    000, 100, 010, 110, 001, 101, 011, 111.
 //    This function converts the statevector from one representation to the other.
+//    This is a special case of reorder_qubits
 // Input: the input statevector and the number of qubits
-// Returns: the statevector in reverse order
+// Output: the statevector in reverse order
```

Comment:
nit: The standatd is to use the: `Args`/ `Returns` wording instead of `Input`/`Output` (that's a minor issue of course).

Reply:
I changed the `Output` to `Returns`, as in other functions. I generally tried to create a standard template for all our function declarations in matrix_product_state. However, I didn't see a standard in other cpp files; the format is rather loose. In particular, I didn't see anyone using `Args`, instead of `Input`, so I didn't change that.